src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java

```diff
@@ -233,34 +245,39 @@ private List<List<Edge>> findNPathsBetweenSwitches(
             }
 
             // Add the lowest weight path becomes the k-shortest path.
-            List<Edge> newBestPath =
+            FindOneDirectionPathResult newBestPath =
                     getBestPotentialKthShortestPath(potentialKthShortestPaths, bestPaths, weightFunction);
-            bestPaths.add(newBestPath);
+            if (newBestPath != null) {
+                bestPaths.add(newBestPath);
+            }
         }
 
         return bestPaths;
     }
 
-    private List<Edge> getBestPotentialKthShortestPath(Set<List<Edge>> potentialKthShortestPaths,
-                                                       List<List<Edge>> bestPaths,
-                                                       WeightFunction weightFunction) {
-        List<Edge> bestKthShortestPath = new ArrayList<>();
-
+    private FindOneDirectionPathResult getBestPotentialKthShortestPath(
+            Set<FindOneDirectionPathResult> potentialKthShortestPaths,
```

Comment:
Please format the method signature in the same style as before.

Reply:
Fixed